baseplate/random.py

```diff
@@ -0,0 +1,57 @@
+"""Extensions to the standard library `random` module."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import random
+
+
+def weighted_choice(items, weight_key):
+    """Choose a random element from a sequence with weighted distribution.
+
+    :param items: A sequence of items to choose from.
+    :param weight_key: A function that takes an
+        item in ``items`` and returns a non-negative integer
+        weight for that item.
+
+    :return: The chosen item.
+
+    :raises: :py:exc:`ValueError` if any weights are negative or there are no
+        items.
+
+    .. testsetup::
+
+        import random
+        from baseplate.random import weighted_choice
+        random.seed(12345)
+
+    An example of usage:
+
+    .. doctest::
+
+        >>> words = ["apple", "banana", "cantelope"]
+        >>> weighted_choice(words, weight_key=lambda word: len(word))
+        'banana'
+
+    """
+    tickets = 0
```

Comment:
I've gone over this implementation thoroughly and can't find any off by one errors, inclusivity / exclusivity issues, or potentially misweighted results.  Thanks for nothing.